### PR TITLE
Use readTrusted for YAML support

### PR DESCRIPTION
**Pull request by jglick**
Amends #3 using https://github.com/jenkinsci/workflow-multibranch-plugin/pull/16.

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,7 +68,7 @@
       <plugin>
         <groupId>org.codehaus.gmaven</groupId>
         <artifactId>gmaven-plugin</artifactId>
-        <version>1.4</version>
+        <version>1.5-jenkins-3</version>
```
Comment (jglick) [inline pom.xml]: Otherwise generated sources get put in a weird structure that confuses NetBeans at least.

File: pom.xml [hunk 2]
```diff
@@ -192,10 +192,9 @@
   
   <properties>
     <jenkins.version>2.0</jenkins.version>
-    <jenkins-test-harness.version>2.7</jenkins-test-harness.version>
+    <jenkins-test-harness.version>2.9</jenkins-test-harness.version>
```
Comment (jglick) [inline pom.xml]: Picks up https://github.com/jenkinsci/jenkins-test-harness/pull/20.

File: pom.xml [hunk 3]
```diff
@@ -192,10 +192,9 @@
...
     <java.level>7</java.level>
     <groovy.version>2.4.6</groovy.version>
-    <workflow.version>2.0</workflow.version>
```
Comment (jglick) [inline pom.xml]: [explanation](https://jenkins.io/blog/2016/04/22/pipeline-2/)

File: src/main/resources/org/jenkinsci/plugins/plumber/PlumberInterpreter.groovy [hunk 4]
```diff
@@ -47,18 +47,7 @@ class PlumberInterpreter implements Serializable {
     }
 
     def fromYaml(String yamlFile) {
-        def yamlText
-
-        // Need to run on some arbitrary node to read the file.
-        // TODO: Find a way to just read that file from the parent flyweight.
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/plumber/PlumberInterpreter.groovy]: When [JENKINS-33273](https://issues.jenkins-ci.org/browse/JENKINS-33273) is implemented, you should get this optimization for free, at least in multibranch projects.

**Comment by jglick** [comment]
Forgot to deploy the dependency snapshot.

**Comment by abayer** [comment]
Jinx - was about to bring that up. =)

**Comment by abayer** [comment]
:bee:

**Comment by jglick** [comment]
Do not forget to review the upstream PR too…

**Comment by jglick** [comment]
@abayer do you plan to merge this?

**Comment by abayer** [comment]
Sorry, doing so now!

**Pull request merged**